python/ray/experimental/workflow/workflow_access.py

```diff
@@ -106,44 +116,77 @@ class WorkflowManagementActor:
 
     def __init__(self, store: "storage.Storage"):
         self._store = store
-        self._workflow_outputs: Dict[str, ray.ObjectRef] = {}
+        self._workflow_outputs: Dict[str, LatestWorkflowOutput] = {}
+        # Cache step output.
+        self._step_output_cache: Dict[str, LatestWorkflowOutput] = {}
```

Comment:
Please add some comments about why we need these two. I feel we can easily mix the purpose of these two things.